cmd/thanos/store.go

```diff
@@ -381,14 +386,25 @@ func runStore(
 
 			level.Info(logger).Log("msg", "initializing bucket store")
 			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
+
+			// This will stop retrying after set timeout duration.
+			initialSyncCtx, cancel := context.WithTimeout(context.Background(), retryTimeoutDuration*time.Second)
```

Comment:
Any reason we don't derive it from `ctx`? 🤔 
```suggestion
			initialSyncCtx, cancel := context.WithTimeout(ctx, retryTimeoutDuration*time.Second)
```

Reply:
Hey @matej-g , that's actually a really good suggestion.
There is no particular reason as to why we didn't use `ctx` 
I will go ahead and make the suggested change :)